diff --git a/ckanext/datastore/db.py b/ckanext/datastore/db.py
index d15d12e..c0a81ef 100644
--- a/ckanext/datastore/db.py
+++ b/ckanext/datastore/db.py
@@ -812,7 +812,8 @@ def _where(where_clauses_and_values):
 
     for clause_and_values in where_clauses_and_values:
         where_clauses.append('(' + clause_and_values[0] + ')')
-        values += clause_and_values[1:]
+        # Ensure all entities are unencoded
+        values += map(lambda v: urllib.unquote(v) if isinstance(v, basestring) else v, clause_and_values[1:])
 
     where_clause = u' AND '.join(where_clauses)
     if where_clause:
