--- ckanext/reclineview/theme/public/vendor/recline/recline.js
+++ ckanext/reclineview/theme/public/vendor/recline/recline.js
@@ -3447,6 +3448,33 @@
     _.bindAll(this, 'render', 'onRecordChanged');
     this.listenTo(this.model.records, 'add remove reset', this.render);
     this.listenTo(this.model.records, 'change', this.onRecordChanged);
+
+    this.listenTo(this.model, 'query:start', function() {
+      self.notify({loader: true, persist: true});
+    });
+
+    this.listenTo(this.model, 'query:done', function() {
+      self.clearNotifications();
+    });
+
+    this.listenTo(this.model, 'query:fail', function(error) {
+      self.clearNotifications();
+      var msg = '';
+      if (typeof(error) == 'string') {
+        msg = error;
+      } else if (typeof(error) == 'object') {
+        if (error.title) {
+          msg = error.title + ': ';
+        }
+        if (error.message) {
+          msg += error.message;
+        }
+      } else {
+        msg = 'There was an error querying the backend';
+      }
+      self.notify({message: msg, category: 'error', persist: true});
+    });
+
     var state = _.extend({
         hiddenColumns: [],
         columnsOrder: [],
@@ -3829,7 +3857,53 @@
 
   hide: function() {
     this.visible = false;
+  },
+
+  notify: function(flash) {
+    var tmplData = _.extend({
+      message: 'Loading',
+      category: 'warning',
+      loader: true
+      },
+      flash
+    );
+    var _template;
+    if (tmplData.loader) {
+      _template = ' \
+        <div class="alert alert-info alert-loader"> \
+          {{message}} \
+          <span class="notification-loader">&nbsp;</span> \
+        </div>';
+    } else {
+      _template = ' \
+        <div class="alert alert-{{category}} fade in" data-alert="alert"><a class="close" data-dismiss="alert" href="#">×</a> \
+          {{message}} \
+        </div>';
+    }
+    var _templated = $(Mustache.render(_template, tmplData));
+    _templated = $(_templated).appendTo($('.controls'));
+    if (!flash.persist) {
+      setTimeout(function() {
+        $(_templated).fadeOut(1000, function() {
+          $(this).remove();
+        });
+      }, 1000);
+    }
+  },
+
+  // ### clearNotifications
+  //
+  // Clear all existing notifications
+  clearNotifications: function() {
+    var $notifications = $('.controls .alert');
+    $notifications.fadeOut(1500, function() {
+      $(this).remove();
+    });
   }
+
 });
 
 })(jQuery, recline.View);


--- ckanext/reclineview/theme/templates/recline_view.html
+++ ckanext/reclineview/theme/templates/recline_view.html
@@ -6,6 +6,7 @@
        data-module-site_url="{{ h.dump_json(h.url_for('/', locale='default', qualified=true)) }}"
        data-module-resource = "{{ h.dump_json(resource_json) }}";
        data-module-resource-view = "{{ h.dump_json(resource_view_json) }}";
+       class="recline-data-explorer"
        >
     <h4 class="loading-dialog">
       <div class="loading-spinner"></div>
