diff --git a/ckanext/datastore/db.py b/ckanext/datastore/db.py
index f21299a..06ae4db 100644
--- a/ckanext/datastore/db.py
+++ b/ckanext/datastore/db.py
@@ -1189,8 +1189,8 @@ def search(context, data_dict):
     _cache_types(context)
 
     try:
-        context['connection'].execute(
-            u'SET LOCAL statement_timeout TO {0}'.format(timeout))
+        # context['connection'].execute(
+        #     u'SET LOCAL statement_timeout TO {0}'.format(timeout))
         return search_data(context, data_dict)
     except DBAPIError, e:
         if e.orig.pgcode == _PG_ERR_CODE['query_canceled']:
@@ -1216,8 +1216,8 @@ def search_sql(context, data_dict):
     _cache_types(context)
 
     try:
-        context['connection'].execute(
-            u'SET LOCAL statement_timeout TO {0}'.format(timeout))
+        # context['connection'].execute(
+        #     u'SET LOCAL statement_timeout TO {0}'.format(timeout))
         results = context['connection'].execute(
             data_dict['sql'].replace('%', '%%')
         )
