diff --git a/ckanext/reclineview/theme/public/vendor/recline/recline.js b/ckanext/reclineview/theme/public/vendor/recline/recline.js
index 95d5c48..7d348b1 100644
--- a/ckanext/reclineview/theme/public/vendor/recline/recline.js
+++ b/ckanext/reclineview/theme/public/vendor/recline/recline.js
@@ -3479,6 +3479,8 @@ my.SlickGrid = Backbone.View.extend({
         columnsOrder: [],
         columnsSort: {},
         columnsWidth: [],
+        columnsTitle: [],
+        columnsToolTip: [],
         columnsEditor: [],
         options: {},
         fitColumns: false
@@ -3607,6 +3609,14 @@ my.SlickGrid = Backbone.View.extend({
       if (widthInfo){
         column.width = widthInfo.width;
       }
+      var titleInfo = _.find(self.state.get('columnsTitle'),function(c){return c.column === field.id;});
+      if (titleInfo){
+        column.name = titleInfo.title;
+      }
+      var toolTipInfo = _.find(self.state.get('columnsToolTip'),function(c){return c.column === field.id;});
+      if (toolTipInfo) {
+          column.toolTip = toolTipInfo.value;
+      }
       var editInfo = _.find(self.state.get('columnsEditor'),function(c){return c.column === field.id;});
       if (editInfo){
         column.editor = editInfo.editor;
