diff --git a/ckan/lib/helpers.py b/ckan/lib/helpers.py
index fd8b642..7bc8bb2 100644
--- a/ckan/lib/helpers.py
+++ b/ckan/lib/helpers.py
@@ -598,7 +598,8 @@ def get_facet_items_dict(facet, limit=None, exclude_active=False):
     facets = sorted(facets, key=lambda item: item['count'], reverse=True)
     if c.search_facets_limits and limit is None:
         limit = c.search_facets_limits.get(facet)
-    if limit is not None:
+
+    if limit:
         return facets[:limit]
     return facets
 
